clang/lib/DPCT/ASTTraversal.cpp

```diff
@@ -8724,699 +8658,95 @@ const ValueDecl *getDecl(const Expr *E) {
   return nullptr;
 }
 
-void EventAPICallRule::findEventAPI(const Stmt *Node, const CallExpr *&Call,
-                                    const std::string EventAPIName) {
-  if (!Node)
-    return;
-
-  if (auto CE = dyn_cast<CallExpr>(Node)) {
-    if (CE->getDirectCallee()) {
-      if (CE->getDirectCallee()->getNameAsString() == EventAPIName) {
-        Call = CE;
-        return;
-      }
-    }
-  }
-  for (auto It = Node->child_begin(); It != Node->child_end(); ++It) {
-    findEventAPI(*It, Call, EventAPIName);
-  }
-}
-
 void EventAPICallRule::handleEventRecord(const CallExpr *CE,
                                          const MatchFinder::MatchResult &Result,
                                          bool IsAssigned) {
-  report(CE->getBeginLoc(), Diagnostics::TIME_MEASUREMENT_FOUND, false);
-  DpctGlobalInfo::getInstance().insertHeader(CE->getBeginLoc(), HT_Chrono);
-  std::ostringstream Repl;
 
   const ValueDecl *MD = getDecl(CE->getArg(0));
   if (!MD)
     return;
-  // Insert the helper variable right after the event variables
   static std::set<std::pair<const Decl *, std::string>> DeclDupFilter;
-  auto &SM = DpctGlobalInfo::getSourceManager();
   std::string InsertStr;
-  if (isInMacroDefinition(MD->getBeginLoc(), MD->getEndLoc())) {
-    InsertStr += "\\";
-  }
-  InsertStr += getNL();
-  InsertStr += getIndent(MD->getBeginLoc(), SM).str();
-  InsertStr += "std::chrono::time_point<std::chrono::steady_clock> ";
-  InsertStr += getTimePointNameForEvent(CE->getArg(0), true);
-  InsertStr += ";";
-  auto Pair = std::make_pair(MD, InsertStr);
-  if (DeclDupFilter.find(Pair) == DeclDupFilter.end()) {
-    DeclDupFilter.insert(Pair);
-    emplaceTransformation(new InsertAfterDecl(MD, std::move(InsertStr)));
-  }
-
-  // Replace event recording with std::chrono timing
-  Repl << getTimePointNameForEvent(CE->getArg(0), false)
-       << " = std::chrono::steady_clock::now()";
-  const std::string Name =
-      CE->getCalleeDecl()->getAsFunction()->getNameAsString();
 
   auto StreamArg = CE->getArg(CE->getNumArgs() - 1);
   auto StreamName = getStmtSpelling(StreamArg);
   auto ArgName = getStmtSpelling(CE->getArg(0));
   bool IsDefaultStream = isDefaultStream(StreamArg);
-  auto IndentLoc = CE->getBeginLoc();
-  auto &Context = dpct::DpctGlobalInfo::getContext();
 
   if (IsAssigned) {
```

Comment:
The only different of the two branch of this `if` is this `StmtStr = "(" + StmtStr + ", 0)";`, so some code can be extract out of the `if.`